packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/content-posts.js

```diff
@@ -5,26 +5,57 @@ import {
 	__experimentalNavigationMenu as NavigationMenu,
 	__experimentalNavigationItem as NavigationItem,
 } from '@wordpress/components';
-import { useDispatch, useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import NavigationEntityItems from '../navigation-entity-items';
 import { MENU_CONTENT_POSTS, MENU_ROOT } from '../constants';
+import ContentNavigationItem from '../content-navigation-item';
+import SearchResults from '../search-results';
+import useDebouncedSearch from '../use-debounced-search';
 
 export default function ContentPostsMenu() {
-	const showOnFront = useSelect(
-		( select ) =>
-			select( 'core' ).getEditedEntityRecord( 'root', 'site' )
-				.show_on_front,
-		[]
+	const {
+		search,
+		searchQuery,
+		onSearch,
+		isDebouncing,
+	} = useDebouncedSearch();
+
+	const { posts, showOnFront, isResolved } = useSelect(
+		( select ) => {
+			const {
+				getEntityRecords,
+				getEditedEntityRecord,
+				hasFinishedResolution,
+			} = select( 'core' );
+			const getEntityRecodsArgs = [
+				'postType',
+				'post',
+				{
+					search: searchQuery,
```

Comment:
I think that this will implicitly search the content of the posts too. My expectation was that the search will only be applied to post titles so as to make it consistent with existing template searching, but this might make sense too. What do you all think? 🤔

Reply:
> My expectation was that the search will only be applied to post titles so as to make it consistent with existing template searching, but this might make sense too.

Same thought here.  While I would expect it the component to search only by titles, returning results for some matching content may be nice!  Although if we go with that we should probably ensure the results are prioritized with those matching titles to show up first.

Reply:
As far as I know, there is no way to search only in the title. Is there? 🤔 We might need to come up with a custom endpoint if we want to search by title only or do the prioritization.

Reply:
I don't think its a bad thing that we search content, its a feature not a bug!  However, I do have a complaint about how it orders things.  The results seem to be ordered based on the date of the post.  So if I am look for 'example', the posts with the word 'example' in their title show up after more recent posts that have the word 'example' in their content:

![Screen Shot 2021-01-27 at 7 30 10 PM](https://user-images.githubusercontent.com/28742426/106072665-8ffe2f00-60d6-11eb-8989-d42e645939d3.png)

I guess my recommendation would be to sort the results such that results with titles corresponding to the search are at the top.

Reply:
Pushed a commit which puts results with titles at the top.